library/core/class.request.php

```diff
@@ -897,6 +897,30 @@ protected function _setRequestArguments($paramsType, $paramsData = null) {
         $this->_RequestArguments[$paramsType] = $argumentData;
     }
 
+    /**
+     * Decode the environment's post depending on content type.
+     *
+     * @param array $post Usually the {@link $_POST} super-global.
+     * @param array $server Usually the {@link $_SERVER} super-global.
+     * @param array $inputFile Usually **php://input** for the raw input stream.
+     */
+    private function decodePost($post, $server, $inputFile) {
```

Comment:
I personally think that we should keep it like so.
We are defaulting arguments like crazy and, in most case, we have more refactor to do later on because of it.